### PR TITLE
docs(tree2): Typos and improvements

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Fixing typos and trying to improve SharedTree's public API documentation.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: experimental/dds/tree2/src/class-tree/tree.ts [hunk 1]
```diff
@@ -26,34 +26,35 @@ export interface ITree extends IChannel {
 	 * Returns a tree that is known to be compatible with the provided schema. The returned tree exposes an API that is schema-aware.
 	 *
 	 * @remarks
-	 * If the tree is uninitialized (has no schema and no content), the tree is initialized with the provided `initialTree` and `schema`.
+	 * If the tree is uninitialized (has no schema and no content), the tree is initialized with the `initialTree` and
+	 * view `schema` in the provided `config`.
```
Comment (alexvy86) [inline experimental/dds/tree2/src/class-tree/tree.ts:30]: I'm not sure I get why we call it "view schema"; it seems to be used for more than just viewing purposes.

File: experimental/dds/tree2/src/class-tree/tree.ts [hunk 2]
```diff
@@ -26,34 +26,35 @@ export interface ITree extends IChannel {
...
+	 * TODO: Support adapters for handling out-of-schema data.
 	 *
 	 * Doing initialization here, regardless of `AllowedUpdateType`, allows a small API that is hard to use incorrectly.
-	 * Other approach tend to have leave easy to make mistakes.
-	 * For example, having a separate initialization function means apps can forget to call it, making an app that can only open existing document,
+	 * Other approaches tend to have easy-to-make mistakes.
+	 * For example, having a separate initialization function means apps can forget to call it, making an app that can only open existing documents,
 	 * or call it unconditionally leaving an app that can only create new documents.
-	 * It also would require the schema to be passed into to separate places and could cause issues if they didn't match.
+	 * It also would require the schema to be passed in to separate places and could cause issues if they didn't match.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/class-tree/tree.ts:51]:
```suggestion
	 * It also would require the schema to be passed into separate places and could cause issues if they didn't match.
```

File: experimental/dds/tree2/src/class-tree/tree.ts (hunk 1 above)
Comment (Josmithr) [reply experimental/dds/tree2/src/class-tree/tree.ts:30]: @CraigMacomber

**Review by Josmithr — APPROVED** [review]

**Comment by alexvy86** [comment]
Merging for the sake of having the minor fixes; we can keep talking about "view schema" separately.

**Pull request merged**